### PR TITLE
Added recipe for warcio

### PR DESCRIPTION
**Pull request by pmlandwehr**
[`warcio`](https://github.com/webrecorder/warcio) helps with reading and writing WARC files.

### CODE REVIEW

File: recipes/warcio/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  imports:
+    - warcio
+
+  commands:
+    - warcio --help
+
+about:
+  home: https://github.com/webrecorder/warcio
+  # No local license: see https://github.com/webrecorder/warcio/pull/25
+  license_file: {{ RECIPE_DIR }}/LICENSE
```
Comment (jakirkham) [inline recipes/warcio/meta.yaml]: xref: https://github.com/webrecorder/warcio/pull/25

File: recipes/warcio/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.{{ bundle }}
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ bundle }}
+  {{ hash_type }}: {{ hash }}
+
+build:
+  noarch: python
+  preserve_egg_dir: True
```
Comment (jakirkham) [inline recipes/warcio/meta.yaml]: Are we sure this is needed?

**Comment by jakirkham** [comment]
Friendly nudge, @pmlandwehr. 😉

File: recipes/warcio/meta.yaml (hunk 2 above)
Comment (pmlandwehr) [reply recipes/warcio/meta.yaml]: It was generated by skeleton, so I've kept it. Not sure what the right test is for validating this?

Comment (dougalsutherland) [reply recipes/warcio/meta.yaml]: I *think* you only need this if you have non-`console_scripts` entry points, which warcio [doesn't](https://github.com/webrecorder/warcio/blob/594eb8c8b0fb35e03294635d23146160832b4371/setup.py#L41).

**Pull request closed without merging**